clippy_lints/src/types/mod.rs

```diff
@@ -264,6 +264,57 @@ declare_clippy_lint! {
     /// struct Foo {
     ///     inner: Rc<Vec<Vec<Box<(u32, u32, u32, u32)>>>>,
     /// }
+    ///
+    /// fn main() {
+    ///     let inner_data: Vec<Vec<Box<(u32, u32, u32, u32)>>> = vec![
+    ///         vec![
+    ///             Box::new((1, 2, 3, 4)),
+    ///             Box::new((5, 6, 7, 8)),
+    ///         ],
+    ///         vec![
+    ///             Box::new((9, 10, 11, 12)),
+    ///         ],
+    ///     ];
+    ///
+    ///     let rc_inner_data: Rc<Vec<Vec<Box<(u32, u32, u32, u32)>>>> = Rc::new(inner_data);
+    ///
+    ///     let foo = Foo {
+    ///         inner: rc_inner_data,
+    ///     };
+    ///
+    ///     // ...
+    /// }
+    /// ```
+    /// Use instead:
+    /// ### Example
+    /// ```no_run
+    /// # use std::rc::Rc;
+    /// type InnerType = Vec<Vec<Box<(u32, u32, u32, u32)>>>;
```

Comment:
Nit: The example could be even more helpful if the types were named by what they supposedly represent, instead of having purely technical names. `PointMatrix` (or whatever) instead of `InnerType`, and `SharedPointMatrix` instead of `RcInnerType`. Then `let shared_matrix: SharedPointMatrix = Rc::new(inner_data)`.